### PR TITLE
adding static-path

### PR DESCRIPTION
**Pull request by nishanthmahasamudram**
Initial PR for implementing AFER DM Caseload, adding [static-path](https://www.npmjs.com/package/static-path?activeTab=readme) and hmppsRestClient.

### CODE REVIEW

File: server/routes/assessingLicences/handlers/caseload.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { Request, Response } from 'express'
+import DecisionMakerCaseloadService from '../../../services/decisionMakerCaseloadService'
+
+export default class CaseloadRoutes {
+  constructor(private readonly decisionMakerCaseloadService: DecisionMakerCaseloadService) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const caseload = await this.decisionMakerCaseloadService.getDecisionMakerCaseload()
```
Comment (nishanthmahasamudram) [inline server/routes/assessingLicences/handlers/caseload.ts:8]: Will Integrate API in the following PR.

File: .circleci/config.yml [hunk 2]
```diff
@@ -71,8 +71,9 @@ jobs:
 
   integration_test:
     executor:
-      name: hmpps/node
-      tag: << pipeline.parameters.node-version >>
+      name: hmpps/node_redis
```
Comment (nishanthmahasamudram) [inline .circleci/config.yml:74]: enabling redis in circleci

File: docker-compose.yml [hunk 3]
```diff
@@ -1,6 +1,22 @@
 version: '3.1'
 services:
 
+  redis:
```
Comment (nishanthmahasamudram) [inline docker-compose.yml:4]: redis for local env

File: server/routes/caseAdmin/handlers/caseload.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { Request, Response } from 'express'
+import CaseAdminCaseloadService from '../../../services/caseAdminCaseloadService'
+
+export default class CaseloadRoutes {
+  constructor(private readonly caseAdminCaseloadService: CaseAdminCaseloadService) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
```
Comment (nishanthmahasamudram) [inline server/routes/caseAdmin/handlers/caseload.ts:7]: will integrate with API in following PR.

File: server/routes/home/index.ts [hunk 5]
```diff
@@ -1,15 +1,18 @@
 import { RequestHandler, Router } from 'express'
+import { path } from 'static-path'
```
Comment (nishanthmahasamudram) [inline server/routes/home/index.ts:2]: adding static path for routing

File: server/data/hmppsRestClient.ts [hunk 6]
```diff
@@ -0,0 +1,260 @@
+import { Readable } from 'stream'
```
Comment (andrewrlee) [inline server/data/hmppsRestClient.ts:1]: we should already have something for this?

Comment (andrewrlee) [reply server/data/hmppsRestClient.ts:1]: **suggestion:** use [rest client](https://github.com/ministryofjustice/hmpps-template-typescript/blob/main/server/data/restClient.ts) instead and remove multer dependency.

We've got to be a bit careful of copying across old idioms from CVL

Comment (nishanthmahasamudram) [reply server/data/hmppsRestClient.ts:1]: Ya this one: server/data/restClient.ts

File: server/routes/caseAdmin/handlers/caseload.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+    res.render('pages/assessingLicences/caseload')
```
Comment (andrewrlee) [inline server/routes/caseAdmin/handlers/caseload.ts:8]: **suggestion:** update nunjucks path to match route path

File: server/routes/index.ts [hunk 8]
```diff
@@ -3,12 +3,14 @@ import { Router } from 'express'
...
 import homeRoutes from './home'
+import assessingLicencesRoute from './caseAdmin'
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-export default function routes({ auditService }: Services): Router {
+export default function routes(services: Services): Router {
   const router = Router()
   router.use(homeRoutes())
   router.use(supportRoutes())
+  router.use(assessingLicencesRoute(services))
```
Comment (andrewrlee) [inline server/routes/index.ts:13]: **suggestion:** worth renaming this?

File: server/middleware/setUpStaticResources.ts [hunk 9]
```diff
@@ -22,9 +22,19 @@ export default function setUpStaticResources(): Router {
...
     router.use('/assets', express.static(path.join(process.cwd(), dir), cacheControl))
   })
+  ;['/node_modules/jquery/dist/jquery.min.js'].forEach(dir => {
+    router.use('/assets/js/jquery.min.js', express.static(path.join(process.cwd(), dir), cacheControl))
+  })
+  ;['/node_modules/jquery-ui-dist/jquery-ui.min.js'].forEach(dir => {
+    router.use('/assets/js/jquery-ui.min.js', express.static(path.join(process.cwd(), dir), cacheControl))
+  })
+  ;['/node_modules/jquery-ui-dist/jquery-ui.min.css'].forEach(dir => {
+    router.use('/assets/stylesheets/jquery-ui.min.css', express.static(path.join(process.cwd(), dir), cacheControl))
```
Comment (andrewrlee) [inline server/middleware/setUpStaticResources.ts:36]: do we need jquery?

File: server/data/tokenStore/systemToken.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+export type SystemToken = { token: string; expiresIn: number }
+export type SystemTokenSupplier = (username?: string) => Promise<SystemToken>
+
+const authHeaderValue = generateOauthClientToken(
+  config.apis.hmppsAuth.systemClientId,
+  config.apis.hmppsAuth.systemClientSecret,
+)
+
+export const getSystemToken: SystemTokenSupplier = async (username?: string): Promise<SystemToken> => {
```
Comment (andrewrlee) [inline server/data/tokenStore/systemToken.ts:15]: **issue:** this stuff always exists in `hmppsAuthClient`

File: server/middleware/setUpStaticResources.ts (hunk 9 above)
Comment (nishanthmahasamudram) [reply server/middleware/setUpStaticResources.ts:36]: Yes we need that for MOJFrontend.SortableTable and other MOJFrontend stuff.

File: server/services/caseAdminCaseloadService.ts [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+
+export default class CaseAdminCaseloadService {
+  constructor(
+    private readonly assessForEarlyReleaseApiClientBuilder: RestClientBuilder<AssessForEarlyReleaseApiClient>,
+  ) {}
+
+  public async getCaseAdminCaseload(token: string) {
+    const assessForEarlyReleaseApiClient = this.assessForEarlyReleaseApiClientBuilder(token)
+    return assessForEarlyReleaseApiClient.getCaseAdminCaseload()
+  }
```
Comment (andrewrlee) [inline server/services/caseAdminCaseloadService.ts:12]: will need a test for this

File: server/routes/caseAdmin/handlers/caseload.ts (hunk 4 above)
Comment (andrewrlee) [reply server/routes/caseAdmin/handlers/caseload.ts:7]: will need a test for this

**Review by andrewrlee — APPROVED** [review]
LGTM

**Comment by nishanthmahasamudram** [comment]
Will add test in my next API Integration PR as it will be a complete end-to-end flow.

**Pull request merged**